include/eve/detail/function/simd/x86/load.hpp

```diff
@@ -11,125 +11,110 @@
 #pragma once
 
 #include <eve/as.hpp>
+#include <eve/concept/memory.hpp>
 #include <eve/concept/vectorizable.hpp>
 #include <eve/detail/abi.hpp>
-#include <eve/detail/meta.hpp>
-#include <eve/memory/aligned_ptr.hpp>
+#include <eve/detail/category.hpp>
 
 namespace eve::detail
 {
   //================================================================================================
-  // 128 bits loads
+  // Regular loads
   //================================================================================================
-  template<real_scalar_value T, typename N>
-  EVE_FORCEINLINE auto load(eve::as_<wide<T, N>> const &, eve::x86_128_ const &, T const* p)
+  template<scalar_pointer Ptr, typename T, typename N, x86_abi ABI>
+  EVE_FORCEINLINE auto load(eve::as_<wide<T, N>> const &, ABI const &, Ptr p)
+  requires( std::same_as<T, std::remove_cvref_t<decltype(*p)>> )
   {
-    if constexpr( N::value * sizeof(T) == x86_128_::bytes )
+    if constexpr( !std::is_pointer_v<Ptr> )
     {
-      if constexpr( std::is_same_v<T, double> )
-      {
-        return _mm_loadu_pd(p);
-      }
-      else if constexpr( std::is_same_v<T, float> )
-      {
-        return _mm_loadu_ps(p);
-      }
-      else if constexpr( std::is_integral_v<T> )
-      {
-        return _mm_loadu_si128((__m128i *)p);
-      }
+      return load(eve::as_<wide<T, N>>{}, ABI{}, p.get());
```

Comment:
But this will generate an unaligned load, no? Since this is `x86` it's not OK

Reply:
Oh this is an error yes, was too sloppy with my copy pasta. Will fix